### DISCUSSION TITLE
What is DNS Bypassed?

### DISCUSSION THREAD

**Discussion by raven9nine** [celzero/rethink-app#504] [Q&A]
Firewall settings
Universal
Block Connection When DNS is Bypassed (Experimental)

I turned this setting on and in my firewall logs I see connections by Brave web browser to Twitter are blocked. The dialogue has the label  DNS Bypassed. The associated message says, 'This connection was blocked because the source app bypassed user-preferred DNS.'

What does this mean exactly? Is this something to be concerned about?

**Comment by ignoramous** [comment]
`DNS Bypass` is when an app connects to IP addresses without first having it resolved via the DNS setup with Rethink. Browsers may do their own DoH (DNS-over-HTTPS) for example which bypasses DNS a user sets with Rethink; Telegram does this mostly as a censorship-resistance feature. Some apps may connect to IP without relying on DNS altogether (and there are legitimate use cases for this), and this would also be blocked (as there's no associated DNS query for the outgoing IP connection). Apps like Instagram and WhatsApp do this, for whatever reasons. You can `Bypass Universal` such apps.

I hope that made it clear?
